[HanB1n] iP

HanB1n opened the pull request:
# ChatterBox

> “Your mind is for having ideas, not holding them.” – David Allen

ChatterBox frees your mind fo having to remember things you need to do!! It's
- text-base
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/HanB1n/ip/releases/tag/A-Jar).
2. double-click it
3. add your tasks.
4. let it manage your tasks for you! :+1:

And its **FREE**

Features:
- [x] Managing tasks
- [ ]  Managing deadline (in progress)
- [ ]  Reminders (in progress)

If your are a Java programmer, you can use it to practice Java too. Here's the main method:
_____________________________________________________________________________________
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(GraphicUi.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

HanB1n committed:
Level-0

HanB1n committed:
Level-0

HanB1n committed:
Level-0

HanB1n committed:
Level-0

HanB1n committed:
Level-1

HanB1n committed:
Level-2

HanB1n committed:
Level-3

HanB1n committed:
Level-3

HanB1n committed:
Level-4

HanB1n committed:
A-TextUiTesting

HanB1n committed:
Level-5

HanB1n committed:
Level-6

HanB1n committed:
Level-6

HanB1n committed:
Add save feature

Implemented Enums
Implemented save feature for the task list using a fixed file in the
current working directory to allow task list to be carried over
multiple runs of the application.

HanB1n committed:
Level-7

Implement OS-independent save feature.
Save file is saved in a directory and works independent of OS.

HanB1n committed:
Level-8

Implement the support of date time for all time based task.
Implement the support of saving date time in the save file

HanB1n committed:
Make the code more OOP

Extract Ui to deal with user interactions.
Extract Storage to deal with saving and loading task from save file.
Extract Parser to handle user inputs.
TaskList completed before
Renamed ToDos to ToDo for standardisation.

HanB1n committed:
A-Package

Divide classes into packages
Enums, Parser, Ui, StoredList, Storage packaged as Utils
All Errors are packaged as ChatterBoxErrors
All types of Tasks are packaged as Tasks

HanB1n committed:
Merge branch 'add-gradle-support'

HanB1n committed:
A-Junit

Implement test for Storage
Implement test for Parser

HanB1n committed:
A-JavaDoc

Add JavaDocs for all non-trivial methods in all Classes.
Add JavaDocs for all Exceptions.

HanB1n committed:
A-CodingStandard

Tweak the code to comply with the course coding standard.

HanB1n committed:
Level-9

Implement find function to search for tasks.

HanB1n committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/ChatterBoxErrors/ChatterBoxInvalidDateError.java
#	src/main/java/ChatterBoxErrors/ChatterBoxInvalidDateTimeError.java
#	src/main/java/Tasks/Deadline.java

HanB1n committed:
Resolve Merge Conflicts

Resolve merge conflict from branch-A-CodingStandard.
Revolve merge conflict from branch-Level-9.
Resolve merge conflict from branch-JavaDoc.

HanB1n committed:
style: Add checkstyle support

Checkstyle support for gradle implemented
Checkstyle uses CS2103T Addressbook checkstyle

Package ChatterBoxErrors renamed to chatterboxerrors

Checkstyle applied to all classes

wujiaweijoanna commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+    public Deadline(String name, LocalDateTime by) {
+        super(name);
+        this.endTime = by;
+
```
Maybe you can delete this empty line?

wujiaweijoanna commented on src/test/java/utils/ParserTest.java:
```diff
@@ -0,0 +1,197 @@
+import chatterboxerrors.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Maybe you can list the imported classes explicitly?

wujiaweijoanna commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+     * @return Save format for the task.
+     */
+    @Override
+    public String storeTask() {
```
The naming in this PR aligns well with the established conventions 👍

wujiaweijoanna commented on src/main/java/Chatterbox.java:
```diff
@@ -0,0 +1,125 @@
+        }
+    }
+
+    /**
```
All of the Javadoc comments are clear and well-written, making the code easy to understand and maintain 🎉

wujiaweijoanna reviewed (commented):
LGTM. 🏆
Just a few nits to fix.

maximustan2 commented on src/main/java/Chatterbox.java:
```diff
@@ -0,0 +1,125 @@
+/**
+ * Represents a Chatbot.
+ */
+public class Chatterbox {
```
Perhaps camelCase could be used for the classname

maximustan2 commented on src/main/java/chatterboxerrors/ChatterBoxDataFileError.java:
```diff
@@ -0,0 +1,26 @@
+     */
+    public ChatterBoxDataFileError() {
+        super("""
+              ____________________________________________________________
```
Perhaps you could consider letting all the separators for all the files be printed from a common utility function or final value to make the code cleaner!

maximustan2 commented on src/main/java/utils/Commands.java:
```diff
@@ -0,0 +1,13 @@
+package utils;
+
+public enum Commands {
```
I like how enums were used here! 🎉

maximustan2 commented on src/test/java/utils/ParserTest.java:
```diff
@@ -0,0 +1,197 @@
+package utils;
+
+import chatterboxerrors.*;
```
Its better not to use wildcard imports as it violates the coding standard

HanB1n committed:
feature Graphical User Interface

Replace Text Based UI with new Graphical UI.

Rename Ui to TextUi for differentiation with new GraphicalUi.

Add resource for dialog pictures.

Fix bug where Parser does not throw relevant error for unknown
command.

maximustan2 reviewed (commented):
LGTM! Just some minor nits

HanB1n committed:
Include assertions

Add assertion to ensure vital assumptions hold in operations.

The various commands in has assertions to ensure there is always
the adequate number of details in every case.

Read from save would always check and create a save file if not
found before the operation of the chatbot to ensure there would be
a save file to write to.

Delete function would check to ensure access to task is bounded by
the size so no deletes can occur with index larger than the size of
task list.

HanB1n committed:
Code Quality Change

Magic numbers and strings are removed in various classes.
Remove unused methods and variables.

HanB1n committed:
Merge branch branch-A-CodeQuality

Resolve conflicts found in StoredList.

HanB1n committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/utils/StoredList.java

HanB1n committed:
New Feature Reminder

Chatterbox reminds of upcoming task with times.

Chatterbox reminds of overdue task with times.

Graphical User Interface support for reminders.
Show reminder on application start up.

HanB1n committed:
Add a User Guide

A user guide is created with Quick Start guide.

All functions and their explanations are found in the User Guide.

Example of GUI is provided in the docs.

Example of start up GUI is added to the User Guide.

HanB1n committed:
Update README

User Guide formatting errors corrected.

Updated project readme.

HanB1n committed:
Update Project README

Fixed project readme where the start up preview of the application
cannot be loaded.

The link to startup.png is edited to accurately reference the
image.

HanB1n committed:
Code readability improvements

Make code of test class more readable.
Refractor variables that are used multiple times.

Edited import statements to fit java coding standards.

Move test_save.txt into test directory for project management.
